pkg/operator/k8sutil/service.go

```diff
@@ -92,3 +98,72 @@ func ParseServiceType(serviceString string) v1.ServiceType {
 	}
 	return v1.ServiceType("")
 }
+
+// ExportService exports a k8s service using Multi Cluster Service API client
+func ExportService(ctx context.Context, client *mcsv1Client.Clientset, service *v1.Service) error {
+	serviceExport := &mcsv1a1.ServiceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            service.Name,
+			Namespace:       service.Namespace,
+			OwnerReferences: service.GetOwnerReferences(),
+		},
+	}
+
+	_, err := client.MulticlusterV1alpha1().ServiceExports(service.Namespace).Create(ctx, serviceExport, metav1.CreateOptions{})
+	if err != nil {
+		if kerrors.IsAlreadyExists(err) {
+			logger.Infof("exported service is already created for the service %q", service.Name)
+			return nil
+		}
+		return errors.Wrapf(err, "failed to create exported service %q", service.Name)
+	}
+
+	return nil
+}
+
+// VerifyExportedService verifies if the ServiceExport status conditions to determine if the service was exported correctly
+func VerifyExportedService(ctx context.Context, client *mcsv1Client.Clientset, name, namespace string) error {
+	retryCount := 5
+	serviceExportWaitTime := 10 * time.Second
+
+	for i := 0; i < retryCount; i++ {
+		exportedService, err := client.MulticlusterV1alpha1().ServiceExports(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to get exported service %q", name)
+		}
+
+		var serviceExportError error
+		if len(exportedService.Status.Conditions) == 0 {
+			serviceExportError = fmt.Errorf("failed to determine status of the exported service")
+		} else {
+			for _, condition := range exportedService.Status.Conditions {
+				if condition.Status != corev1.ConditionTrue {
+					serviceExportError = fmt.Errorf(*condition.Message)
+				}
+			}
+		}
+
+		if serviceExportError != nil {
+			if i < retryCount-1 {
+				logger.Warningf("failed to verify exported service status for %q. Trying again...", name)
+				time.Sleep(serviceExportWaitTime)
+				continue
+			} else {
+				return errors.Wrapf(serviceExportError, "failed to verify exported service %q", name)
+			}
+		}
+	}
+
+	return nil
+}
+
+func GetExportedServiceIP(ctx context.Context, clientset kubernetes.Interface, name, namespace string) (string, error) {
+	// TODO: get external IP from the DNS query on <service>.<ns>.svc.clusterset.local once https://github.com/submariner-io/lighthouse/issues/993 is fixed
```

Comment:
Yes. This will be part of submariner 0.15 (ACM 2.8). Looks like it was merged. So I can try with upstream submariner and get external IP using DNS query.

Reply:
Using nslookup to resolve dns